doc/reference/plutus/reference-script-example.md

```diff
@@ -0,0 +1,148 @@
+# Plutus Scripts
+
+## What is a reference script?
+
+A reference script is a script that exists at a particular transaction output. It can be used to witness, for example, a UTxO at the corresponding script address of said reference script. This is useful because the script does not have to be included in the transaction anymore, which significantly reduces the transaction size.
+
+### An example of using a Plutus V2 reference script
+
+Below is an example that shows how to use a Plutus spending script. Here we discuss a [shell script example of how to use a reference script to spend a tx input](scripts/plutus/example-reference-script-usage.sh). This is a step-by-step process involving:
+
++ the creation of the `Required Redeemer` Plutus txin script
++ the creation of the `Required Redeemer` Plutus script at a transaction output (creation of the reference script)
++ sending ada and a datum to the Plutus script address
++ spending ada at the Plutus script address using the Plutus reference script
+
+In this example we will use the [Required Redeemer](scripts/plutus/scripts/v2/required-redeemer.plutus) Plutus spending script. In order to execute a reference Plutus spending script, we require the following:
+
+- Collateral tx input(s) - these are provided and are forfeited in the event the Plutus script fails to execute.
```

Comment:
Should we ease folks mind by telling them about the fact that a local node will not forward on a failing tx to the relay node unless it is explicitly told to? As you know, I'm sure, the collateral exists to prevent abusive behavior, and under normal circumstances will not be taken.